### PR TITLE
[RCNN Memory Optimize] Refine memory usage of some operators

### PR DESCRIPTION
**Pull request by sneaxiy**
This PR saves the memory usage of `roi_align_grad`, `sigmoid_cross_entropy_with_logits` and `sigmoid_cross_entropy_with_logits_grad` operators. 

The maximum batch size of Mask-RCNN-R101-FPN-1x increases from 3 to 5.

### CODE REVIEW

**Review by chengduoZH — APPROVED** [review]
LGTM
